### PR TITLE
Add params for delay in k8s

### LINKED ISSUE #927: [infra] Simulate bad network conditions
- Add support to set:

- [x] latency
- [x] packet loss
- [x]  jitter

We should support in the available providers:
`k8s` through chaos-mesh CRD (installed in the cluster)
`native` ( tc & netem on Linux)
`podman` (need exploration).

### PR DESCRIPTION
**Pull request by wirednkod**
Fixes #927

### CODE REVIEW

File: javascript/packages/orchestrator/src/providers/k8s/resources/chaosResource.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return this.generateChaosSpec(name, ports, delay);
+  }
+
+  private shouldExposeJaegerPorts(): boolean {
+    return this.podSpec.spec.containers.some(
+      (container) => container.name === "jaeger-agent",
+    );
+  }
+
+  private generatePorts(): ServiceSpec["spec"]["ports"] {
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/providers/k8s/resources/chaosResource.ts:43]: We can remove this fn.

File: javascript/packages/orchestrator/src/providers/k8s/kubeClient.ts [hunk 2]
```diff
@@ -141,8 +141,8 @@ export class KubeClient extends Client {
 
     await this.createResource(podDef, true);
     if (podDef.metadata.labels["zombie-role"] !== ZombieRole.Temp) {
-      const serviceDef = genServiceDef(podDef);
-      await this.createResource(serviceDef, true);
+      const genFn = podDef.spec.delay ? genChaosDef : genServiceDef;
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/providers/k8s/kubeClient.ts:144]: Is not one or the other, we need to deploy both manifests if the `delay` is set. One to create the service, this allow us to refer to the pod by `dns` and the `chaos` to inject network failures.

File: javascript/packages/orchestrator/src/types.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ export interface Settings {
   backchannel?: boolean; // only used in k8s at the moment, spawn a backchannel instance
   image_pull_policy?: "IfNotPresent" | "Never" | "Always";
   local_ip?: string; // ip used for expose local services (rpc/metrics/monitors)
+  bad_network_global_settings?: BadNetworkSettings;
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/types.ts:40]: If we have this set, we need to `mv` or `copy` this config to all the node when we create the NetworkSpec.

**Comment by pepoviola** [comment]
@wirednkod, I added a new test for our `ci` and fix an issue in the chaos manifest. I think this is ready to go ones the ci finish.

Thanks!

**Comment by pepoviola** [comment]
Hi @wirednkod, I will refactor this since applying the delay when we create the pods cause performance issues with the spawn of the network. I will change the logic to spawn the network and apply all the manifest at once before returning the network instance.

**Comment by wirednkod** [comment]
> Hi @wirednkod, I will refactor this since applying the delay when we create the pods cause performance issues with the spawn of the network. I will change the logic to spawn the network and apply all the manifest at once before returning the network instance.

Thank you @pepoviola

File: javascript/packages/orchestrator/src/providers/k8s/kubeClient.ts [hunk 4]
```diff
@@ -219,6 +220,13 @@ export class KubeClient extends Client {
 
     await this.putLocalMagicFile(name);
     await this.waitPodReady(name);
+
+    // if (podDef.metadata.labels["zombie-role"] !== ZombieRole.Temp && delay) {
+    //   const chaosDef = genChaosDef(name, this.namespace, delay);
+    //   writeLocalJsonFile(this.tmpDir, `${name}-chaos.json`, chaosDef);
+    //   await this.createResource(chaosDef, true);
+    // }
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/providers/k8s/kubeClient.ts:228]: remove comments

**Comment by pepoviola** [comment]
Delay test is now green @wirednkod, I think we can merge this on as is. I will open a new issue to support add/remove `delays` as part of the `zndsl`.
Thx!

**Comment by pepoviola** [comment]
@wirednkod merging now, and I created this followup issue https://github.com/paritytech/zombienet/issues/1302

We should also add the docs for this settings in the spawn phase.

**Pull request merged**